Assets/Xbox Live/Scripts/SignInManager.cs

```diff
@@ -0,0 +1,605 @@
+    {
+        foreach (var playerInfo in this.CurrentPlayers)
+        {
+            if (playerInfo.Value.XboxLiveUser != null && playerInfo.Value.XboxLiveUser.XboxUserId.Equals(xuid))
+            {
+                return playerInfo.Key;
+            }
+        }
+        return -1;
+    }
+
+    /// <summary>
+    /// Returns the Xbox Live Context for the Xbox Live User with the assigned player number.
+    /// </summary>
+    /// <param name="playerNumber">The player number.</param>
+    /// <returns>The Xbox Live Context for the Xbox Live User with the assigned player number.</returns>
+    public XboxLiveContext GetXboxLiveContextForPlayer(int playerNumber)
+    {
+        if (ValidatePlayerNumber(playerNumber, "Get Xbox Live Context", XboxLiveOperationType.GetUser))
+        {
+            return new XboxLiveContext(this.CurrentPlayers[playerNumber].XboxLiveUser);
+        }
+        return null;
+    }
+
+    /// <summary>
+    /// Validates whether the playerNumber if allowed for the operation requested or not
+    /// </summary>
+    /// <returns>True if valid. False, otherwise</returns>
+    private bool ValidatePlayerNumber(int playerNumber, string operationName, XboxLiveOperationType operationType)
+    {
+        if (playerNumber <= 0)
+        {
+            if (XboxLiveServicesSettings.Instance.DebugLogsOn)
+            {
+                Debug.LogError(operationName + " Failed: Player Number needs to be greater than zero.");
+            }
+            return false;
+        }
+
+        if (playerNumber > GetMaximumNumberOfPlayers())
+        {
+            if (XboxLiveServicesSettings.Instance.DebugLogsOn)
+            {
+                Debug.LogError(operationName + " Failed: Player Number exceeds the maximum number of users allowed on this platform.");
+            }
+            return false;
+        }
+
+        if (operationType == XboxLiveOperationType.SignIn)
+        {
+            if (CurrentPlayers[playerNumber].XboxLiveUser != null)
+            {
+                if (XboxLiveServicesSettings.Instance.DebugLogsOn)
+                {
+                    Debug.LogError(operationName + " Failed: Player " + playerNumber + " is already signed in.");
+                }
+                NotifyAllCallbacks(
+                    playerNumber,
+                    null,
+                    XboxLiveAuthStatus.Invalid,
+                    operationName + " Failed: Player " + playerNumber + " is already signed in.",
+                    true);
+                return false;
+            }
+        }
+
+        if (operationType == XboxLiveOperationType.SignOut || operationType == XboxLiveOperationType.GetUser)
+        {
+            if (!CurrentPlayers.ContainsKey(playerNumber) || CurrentPlayers[playerNumber].XboxLiveUser == null)
+            {
+                if (XboxLiveServicesSettings.Instance.DebugLogsOn)
+                {
+                    if (operationType == XboxLiveOperationType.SignOut)
```

Comment:
Redundant if check. Can just print the error message.

Reply:
Done everywhere. Will be in the next push.